src/main/java/seedu/address/logic/commands/EditRestaurantNoteCommand.java

```diff
@@ -0,0 +1,153 @@
+            + ": " + COMMAND_FUNCTION + "\n"
+            + "Parameters: INDEX and LINE_NUMBER (must be a positive integer) "
+            + PREFIX_LINE_NUMBER_RECOMMENDED + "LINE_NUMBER " + PREFIX_RECOMMENDED + "INFO "
+            + PREFIX_LINE_NUMBER_GOOD + "LINE_NUMBER " + PREFIX_GOOD + "INFO "
+            + PREFIX_LINE_NUMBER_BAD + "LINE_NUMBER " + PREFIX_BAD + "INFO\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_LINE_NUMBER_RECOMMENDED + " 2 " + PREFIX_RECOMMENDED + " Lobster pasta "
+            + PREFIX_LINE_NUMBER_GOOD + " 1 " + PREFIX_GOOD + " Mushroom soup "
+            + PREFIX_LINE_NUMBER_BAD + " 3 " + PREFIX_BAD + " Salad";
+
+    public static final String MESSAGE_EDIT_REMARK_SUCCESS = "Edited notes for restaurant: %1$s";
+    public static final String MESSAGE_EMPTY_REC = "No recommended food to be edited is provided.";
+    public static final String MESSAGE_EMPTY_GOOD = "No good food to be edited is provided.";
+    public static final String MESSAGE_EMPTY_BAD = "No bad food to be edited is provided.";
+
+    private final Index index;
+    private final int lineRec;
+    private final int lineGood;
+    private final int lineBad;
+    private final Note recommendedFood;
+    private final Note goodFood;
+    private final Note badFood;
+
+    /**
+     * @param index of the person in the filtered restaurant list to edit the note
+     * @param lineRec number of a specific note in the recommended food information stored
+     * @param lineGood number of a specific note in the good food information stored
+     * @param lineBad number of a specific note in the bad food information stored
+     * @param recommendedFood of the restaurant to be updated to
+     * @param goodFood of the restaurant to be updated to
+     * @param badFood of the restaurant to be updated to
+     */
+    public EditRestaurantNoteCommand(Index index, int lineRec, int lineGood,
+                                     int lineBad, Note recommendedFood, Note goodFood, Note badFood) {
+        requireAllNonNull(index, lineRec, lineGood, lineBad, recommendedFood, goodFood, badFood);
+
+        this.index = index;
+        this.lineRec = lineRec;
+        this.lineGood = lineGood;
+        this.lineBad = lineBad;
+        this.recommendedFood = recommendedFood;
+        this.goodFood = goodFood;
+        this.badFood = badFood;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Restaurant> lastShownList = model.getFilteredRestaurantList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_RESTAURANT_DISPLAYED_INDEX);
+        }
+
+        Restaurant restaurantToEdit = lastShownList.get(index.getZeroBased());
+        if (restaurantToEdit.getRecommendedFood().size() == 0) {
+            throw new CommandException(Messages.MESSAGE_NO_RECOMMENDED_FOOD);
```

Comment:
This doesn't matter right if the user is not trying to edit the saved recommended food? O: